src/sources/datadog/agent/logs.rs

```diff
@@ -105,17 +105,59 @@ pub(crate) fn decode_log_body(
-                            log.try_insert(path!("timestamp"), timestamp);
-                            log.try_insert(path!("hostname"), hostname.clone());
-                            log.try_insert(path!("service"), service.clone());
-                            log.try_insert(path!("ddsource"), ddsource.clone());
-                            log.try_insert(path!("ddtags"), ddtags.clone());
-                            log.try_insert(
+                            let namespace = &source.log_namespace;
+                            let source_name = "datadog_agent";
+
+                            namespace.insert_source_metadata(
+                                source_name,
+                                log,
+                                path!("status"),
+                                status.clone(),
+                            );
+                            namespace.insert_source_metadata(
+                                source_name,
+                                log,
+                                path!("timestamp"),
+                                timestamp,
+                            );
+                            namespace.insert_source_metadata(
+                                source_name,
+                                log,
+                                path!("hostname"),
+                                hostname.clone(),
+                            );
+                            namespace.insert_source_metadata(
+                                source_name,
+                                log,
+                                path!("service"),
+                                service.clone(),
+                            );
+                            namespace.insert_source_metadata(
+                                source_name,
+                                log,
+                                path!("ddsource"),
+                                ddsource.clone(),
+                            );
+                            namespace.insert_source_metadata(
+                                source_name,
+                                log,
+                                path!("ddtags"),
+                                ddtags.clone(),
+                            );
+
+                            namespace.insert_vector_metadata(
+                                log,
                                 path!(source.log_schema_source_type_key),
+                                path!("source_type"),
                                 Bytes::from("datadog_agent"),
                             );
-                            log.try_insert(path!(source.log_schema_timestamp_key), now);
+                            namespace.insert_vector_metadata(
+                                log,
+                                path!(source.log_schema_timestamp_key),
+                                path!("ingest_timestamp"),
+                                now,
+                            );
```

Comment:
For the new `Vector` namespace, do we want the `source_type` and `timestamp` keys to be configurable?

Reply:
Right now these are not configurable. The main reason for being able to configure them previously was to avoid collisions with other fields of the same name. With the `Vector` namespace, these collisions aren't possible (and allowing arbitrary names could end up allowing collisions again).

Reply:
Ah, good. I was worried that they _were_ configurable, rather than not.

So it's not possible to set `source.log_schema_source_type_key` or `source.log_schema_timestamp_key` when using the Vector namespace?

Reply:
You can still set it, but it will be ignored when the `Vector` namespace is used. This will be added to the public docs once the Log Namespace feature is announced. Eventually when the `Legacy` namespace is deprecated / removed those config options will also be removed.